frontend/src/utils/api.ts

```diff
@@ -290,10 +298,10 @@ window.api = {
   setContributionOfCommitResultsAndInsertRepoId(dailyCommits, repoId) {
```

Comment:
I think we can type the parameters here so that we don't need to cast to the specific types that we need within the function.

Reply:
The functions are all typed in `types/window.ts` - for this function we're changing the type of each commit object since we're adding three fields (`repoId`, `insertions`, `deletions`), so the typecast is necessary (I think? not sure how else to do it in Typescript).

I decided to define the two types separately (`CommitResultRaw` = the raw form obtained from the json file without the above 3 fields, `CommitResult` = after adding the three fields) instead of using just one type with the three fields as optional fields. This was to avoid having to check `if (commitResult.repoId)` etc in the rest of the app.

Reply:
Ah yes. Sorry I missed the declaration in `types/window.ts`. I get what you mean now. Yes this looks good to me.